### PR TITLE
Remove stylelint-order until strong course of action is found

### PR DESCRIPTION
**Pull request by cshold**
There's healthy debate about how to organize CSS. Some say based on "like-minded" properties (height/width, position/left/right, etc), some like declaring `display: flex` before `align-items`, and others like alphabetical (they're joking, I'm pretty sure of it). That is to say there is no single way that every FED at Shopify writes CSS. 

I had to revise a [PR](https://github.com/Shopify/assets/pull/817#pullrequestreview-52730571) three times today (because I don't have in-editor linting setup). These are the CI errors I got after each push.
1. `box-sizing` has to go above `padding`
2. `box-sizing` has to go below `position`
3. `box-sizing` has to go below `z-index`

WTF.

What is common is we want to write CSS quickly in an _organized_ way. I don't believe this fixed order is the right way.

### CODE REVIEW

**Comment by lemonmade** [comment]
Can we investigate other orders that we feel are "more correct"? These were a first stab at it, it is absolutely the wrong decision to kill it outright. There are tons of options for how to group these, and how strictly to order within groups.

**Comment by harisaurus** [comment]
I'd be happy to do rule grouping (all paddings together etc), but linting the order of groups or items within groups doesn't feel necessary or useful.

**Comment by tylerball** [comment]
Would it be easier if you had linting set up in your editor? We should _absolutely_ provide links/resources for setting this up in our popular editors, maybe even in the error messaging.

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,11 +2,14 @@
 
 ## [Unreleased]
 
+* Removed plugin
+  * Removed `stylelint-order` because it adds unnecessary overhead to writing CSS
```
Comment (kaelig) [inline CHANGELOG.md]: > because it adds unnecessary overhead to writing CSS

Or… "until we find a consensus on the ordering"?

Comment (cshold) [inline CHANGELOG.md]: True, I did show my bias here a bit.  Happy to update before/if this merges based on the consensus.

**Comment by kaelig** [comment]
@ismail-syed can you take this on?

We need to have a constructive conversation around ordering before switching it on for everyone, until then let's disable it.

Sidenote: Would a patch release make sense?

**Comment by lemonmade** [comment]
I strongly disagree that we should disable. The projects using this are largely OK with the linting. We should have this discussion before this is something we are telling everyone to use, but until then I see no reason to remove.

**Comment by kaelig** [comment]
Are projects that temporarily disable that rule likely to turn it back on?

We need to be mindful of our reputation and how other teams trust us to do the right thing. In this case, we didn't properly sit down and have a solid discussion on the order – it's hard for me to argue for keeping the rule in at this time.

**Comment by kaelig** [comment]
@cshold can you please edit the changelog (see comment)? This will help us get this in and you can resume writing CSS without being interrupted by order rules for a bit.

**Comment by ismail-syed** [comment]
> @ismail-syed can you take this on?

Sorry, what is it exactly what you'd like me to take on? 

I've spent more time on `stylint-config-shopify` this week than I originally anticipated. It might have to wait until next week.

**Comment by kaelig** [comment]
I think it's just a matter of reviewing, merging and publishing this as a patch. If this takes too much time, I can help.

**Comment by ismail-syed** [comment]
I don't think removing this lint rule completely is the right approach. We should take this conversation in an issue. 

Then with all the proposed options (what should/shouldn't be enforced, why enforce it etc) we can have a consensus around what the options should be for ordering.

**Comment by ismail-syed** [comment]
@cshold, can you please create an issue for this so we can have a conversation about this there.

**Comment by hudochenkov** [comment]
stylelint recently added [`--fix` option](https://stylelint.io/user-guide/cli/#autofixing-errors). And `stylelint-order` [supports](https://github.com/hudochenkov/stylelint-order#autofixing) it. So you can automatically sort your stylesheets to desired order.

**Comment by kaelig** [comment]
Closing this as we're getting close to a resolution in #10.

**Pull request closed without merging**